### DISCUSSION TITLE
Non responsive device or Alexa Routines not working.

### DISCUSSION THREAD

**Discussion by Rumbaar** [vintlabs/fauxmoESP#184] [Q&A]
I've used this project to connect a ESP8266 that I'm using the control some transistors to activate some dumb switches.  So it's a light pretending to be a switch.

Now I can set them up and they show up in Alexa, manually turn them on and it works.   But at times the devices disapear and show in Alexa as 'devices doesn't support response" (or something to that affect).  I've also setup a routine that triggers this switches, but when I activate them, nothing happens.

Is this a limitation to the code in regards to me using this on a non-light device?   Using the default example, it toggles ON/OFF states, how can I get it to just toggle ON every time and set itself to OFF immediately?

**Comment by Woose** [comment]
I just set up a system to open and close my garage door using a simple relay connected to a WEMOS D1 (esp8266).  The 'button' on the Alexa app is sporadic, it flips itself to 'off' a few seconds after you press it, 'on'.  And I get that message on the Alexa app, "devises doesn't support requested value".

HOWEVER.  I created a Routine, "Alexa, Open the Garage Door" and "Alexa, Close the Garage Door" and this works GREAT.

My system also has an HC-SR04 ultrasonic distance sensor to see if the door is currently open or shut.

**Reply by Rumbaar** [reply]
Thanks for the reply, the button within Alexa works fine when I activate it and doesn't switch to off (which I want too) and I can't seem to get routine to work via Voice or via directly activating them within Alexa.

Did you use the current versions of the project?

**Comment by Woose** [comment]
When I started the software wouldn’t work at all on an ESP32 or an ESP8266. I deleted the fauxmo library from my libraries then downloaded the fresh on from GitHub. 

After loading the simple example from the FauxmoESP library all 5 ‘colors’ sort of worked.  It takes Alexa some time to sort them out. Be patient. 

That’s all I know!

Frank
Cell (214) 477 4722

**Comment by Steve-Newall** [comment]
I have the same issue with Alexa stating "Device doesn't support requested value".  I have multiple ESP8266 D1 mini devices using fauxmoESP.  Some control a single "light" while others control multiple "lights".   Alexa can turn all of the devices on and off without any problems.  HOWEVER, periodically Alexa asks for the status of the device, and the response from the Alexa app on an iPhone is "Device doesn't support requested value".

I've traced the communications to / from the fauxmoESP devices using WireShark.  I can see the HTTP request to turn on a device and the fauxmoESP HTTP response.  So far so good.  However, after around 1 to 2 seconds there is an Alexa request (HTTP GET) for the device status and the fauxmoESP device formats and sends what looks like a correctly formatted HTTP/JSON response.  However, Alexa apparently doesn't like the response as when using the iPhone Alexa app to turn the "light" on, the light turns on but as noted above, after 1 to 2 seconds the error"Device doesn't support requested value" shows on the iPhone app.

The above is repeatable on multiple ESP8266 devices running fauxmoESP code.  Is there any documentation on what the response message for a device status request SHOULD look like?   I've reviewed the fauxmoESP's templates.h file.  This file defines both the response to send after receiving a turn on/off request, and the response to send after receiving a device status request.

Can anyone suggest what I can try next?  Would the WireShark logs help anyone?

**Reply by Steve-Newall** [reply]
Minor update:  I've looked at the Philips Hue developer information ( sign up required at https://developers.meethue.com/develop/hue-api/lights-api/ ) and from the information on the API to get light attributes and state, the fauxmoESP response looks correct.  I even tried tweaking the response by updating the FAUXMO_DEVICE_JSON_TEMPLATE structure and the code that fills it in to more closely match the Philips hue sample response.  Still get the same issue.  Alexa voice commands control the device without issues, the iPhone Alexa controls the device, even using a Bash based hue-shell controls the devices, but when looking at the WireShark messages and the iPhone Alexa app, after a few seconds the Alexa app says  "Device doesn't support requested value".

One interesting point from the WireShark logs, is that although the ESP8266 / fauxmoESP request to response time is almost instantaneous (less than 1,000th of a second) , I can see a the starting of a second TCP 3-way handshake just as the ESP8266/fauxmoESP is starting to transmit it's 1st response.  This second request is from the SAME source address and is a request for the status of the SAME light.  Go figure.  Seems more like an Alexa issue perhaps?

**Comment by Tejas-MD** [comment]
Does anyone think this is related to this latest problem in Discussion #169[](https://github.com/vintlabs/fauxmoESP/discussions/169)? 

Cause fixing that is highly essential, it's super annoying when it takes twice to make Alexa turn the lights off/on

**Reply by Miq1** [reply]
I have to join the club I am afraid. I heavily modified the Fauxmo code to fix some bugs and get it more stable. I am pretty sure the ``lights`` response is and was syntactically correct according to the Philips Hue developer docs. It would be most interesting what a "real" Hue lamp does respond to see a difference. Does someone have one at hand and can provide a Wirshark log or such?

My Alexa request and my response btw are looking like
```
[N] 811897| hucomm.cpp           [ 430] _onTCPRequest: Request: 1
[N] 811898| hucomm.cpp           [ 431] _onTCPRequest: URL: /api/483fda7e48340000082581facebeadLEDlite000/lights/1
[N] 811903| hucomm.cpp           [ 432] _onTCPRequest: Body:

[N] 811909| hucomm.cpp           [ 133] _sendTCPResponse: TCP Response:
HTTP/1.1 200 OK
Content-Type: application/json
Content-Length: 489
Connection: close

{
  "type":"Extended color light",
  "name": "Tint1",
  "uniqueid": "48:3f:da:7e:48:34:00:00-00",
  "modelid": "LCT015",
  "manufacturername": "Philips",
  "productname": "E4",
  "state":{
    "on": true,
    "bri": 0,
    "effect": "none",
    "alert": "none",
    "hue": 65535,
    "sat": 0,
    "xy": [0.0,0.0],
    "colormode": "ct",
    "ct": 500,
    "mode": "homeautomation",
    "reachable": true
  },
  "capabilities":{
    "certified": false
  },
  "swversion": "5.105.0.21169"
}
```

As described above, Alexa will repeat the request several times, then flag the device as unresponsive.
